### ISSUE TITLE
Automap not working with private fields when only getter exists

### ISSUE THREAD

**Issue by notmedia** [nartc/mapper#281]
In some case I have a class for example:

```ts
class Entity {
  private _uuid: string;

 @AutoMap()
 get uuid() {
   return this._uuid;
 }
}
```

The automap fails, because under the hood automapper try to write metadata to this private field, but the setter does not exist.

I can remove `@AutoMap` decorator and create mapper like this and it would work:

```ts
mapper
  .createMap(Entity, EntityView)
  .forMember(
    (destination) => destination.uuid,
    mapFrom((source) => source.uuid)
  )
....
```
But it's looking terrible when you have many fields, and looks like unnecessary.
Can we achieve this or not without setting setter and unnecessary code?

**Comment by nartc** [comment]
@notmedia a little update, I have made some progress on this with `Proxy`. Not sure if it's a correct approach but Proxy#set trap is probably the only way I found that it works with only-getters. Can you double-check on the browser supports for Proxy as well as Node support?

**Comment by notmedia** [comment]
@nartc Hi! As I see on https://caniuse.com/proxy `Proxy` support in all modern browsers except IE (as usual😬). What browsers we need to test?

I read about `Proxy` functionality and I think this can be used, but if you want to proxy setters with this thing I think that programatically we must don't have access to this setters in our code. I didn't see you code implementation and I just hypothetically imagine this.

It's important, because the case with not defining setters is for OOP patter, where programmer can't change value outside the class.

**Comment by nartc** [comment]
@notmedia It's important to note that classes (models) with only-getters should only be used as `Source`. However, upon initialization with `createMap()`, I do instantiate both `Source` and `Destination` and assign `undefined` properly so I have the objects with properties in them down the stream.

```ts
class Foo {
   foo: string;
   bar: number;
}

const foo = new Foo(); // Foo {} , no "foo", no "bar"
const foo = instantiate(Foo); // Foo { foo: undefined, bar: undefined }
```

So AutoMapper does in-fact try to set things. The approach I'm going with is to set a `useProxy` flag on models that have Only-Getters. If `useProxy` is true, I'll run the instantiation via a `Proxy`.

**Comment by notmedia** [comment]
In my case it's okay, because my classes without setters are Domain/DB models. With this models I work in my service layer. `automapper` used to map my models (`source`) to presentation models (`destination`).

**Comment by nartc** [comment]
Thanks for the inputs. I'll probably push a release either later tonight or tomorrow.

**Comment by nartc** [comment]
@notmedia this should be fixed in 3.5.0 Please give it a try

**Comment by notmedia** [comment]
@nartc Yep, it works! Thank you 👍🏻
